pkg/grafana/grafana.go

```diff
@@ -102,6 +139,117 @@ func DeleteByID(ctx context.Context, grafanaAPI *client.GrafanaHTTPAPI, id int64
 	return nil
 }
 
+func ConfigureDefaultDatasources(ctx context.Context, grafanaAPI *client.GrafanaHTTPAPI, organization Organization) ([]Datasource, error) {
+	logger := log.FromContext(ctx)
+	// TODO using a serviceaccount later would be better as they are scoped to an organization
+
+	var err error
+	// Switch context to the current org
+	if _, err = grafanaAPI.SignedInUser.UserSetUsingOrg(organization.ID); err != nil {
+		logger.Error(err, "failed to change current org for signed in user")
+		return nil, errors.WithStack(err)
+	}
+
+	// We always switch back to the shared org
+	defer func() {
+		if _, err = grafanaAPI.SignedInUser.UserSetUsingOrg(SharedOrg.ID); err != nil {
+			logger.Error(err, "failed to change current org for signed in user")
+		}
+	}()
+
+	configuredDatasourcesInGrafana, err := listDatasourcesForOrganization(ctx, grafanaAPI)
+	if err != nil {
+		logger.Error(err, "failed to list datasources")
+		return nil, errors.WithStack(err)
+	}
+
+	datasourcesToCreate := make([]Datasource, 0)
+	datasourcesToUpdate := make([]Datasource, 0)
+
+	// Check if the default datasources are already configured
+	for _, defaultDatasource := range defaultDatasources {
+		found := false
+		for _, configuredDatasource := range configuredDatasourcesInGrafana {
+			if configuredDatasource.Name == defaultDatasource.Name {
+				found = true
+
+				// We need to extract the ID from the configured datasource
+				datasourcesToUpdate = append(datasourcesToUpdate, defaultDatasource.withID(configuredDatasource.ID))
```

Comment:
Maybe add a `break` here, so we do not continue the loop as we already found match. This assumes a default datasource can only be found once in configured datasources and should only be updated once.

Reply:
Worse case scenario, this loops over 3 items so using a break is not really useful but sure

Reply:
Done